llvm/lib/Analysis/Lint.cpp

```diff
@@ -401,6 +403,10 @@ void Lint::visitMemoryReference(Instruction &I, const MemoryLocation &Loc,
         "Unusual: Address one pointer dereference", &I);
 
   if (Flags & MemRef::Write) {
+    if (TT.isAMDGPU())
+      Check(!AMDGPU::isConstantAddressSpace(UnderlyingObject->getType()->getPointerAddressSpace()),
```

Comment:
Is it possible or better to do the check via TTI? I'm not sure if it's a good idea to do the check based on triple.

Reply:
Just to be clear, the "check" I meant here was `AMDGPU::isConstantAddressSpace`.